src/components/reusable/progressBar/progressBar.ts

```diff
@@ -0,0 +1,273 @@
+  @property({ type: String })
+  status: 'indeterminate' | 'active' | 'success' | 'error' = 'indeterminate';
+
+  /** Sets initial progress bar value (optionally hard-coded). */
+  @property({ type: Number })
+  value: number | null = null;
+
+  /** Sets manual max value (default = 100). */
+  @property({ type: Number })
+  max = 100;
+
+  /** Sets optional progress bar label. */
+  @property({ type: String })
+  label = '';
+
+  /** Sets optionally displayed information icon and tooltip content. */
+  @property({ type: String })
+  informationalTooltipText = '';
+
+  /** Sets optional helper text that appears underneath `<progress>` element. */
+  @property({ type: String })
+  helperText = '';
+
+  /** Sets the unit for progress measurement (ex: 'MB', 'GB', '%') */
+  @property({ type: String })
+  unit = '';
+
+  /** Incrementing percentage count value.
+   * @internal
+   */
+  @state()
+  private _percentage = 0;
+
+  /** Increments animated movement in progress bar.
+   * @internal
+   */
+  @state()
+  private _progress = 0;
+
+  /** Value indicates whether or not the bar is in animated motion, dynamic helper text values are proportionally incrementing.
+   * @internal
+   */
+  @state()
+  private _running = false;
+
+  /** Controls timeout interval for incremented bar animation.
+   * @internal
+   */
+  private _animationFrameId: number | null = null;
+
+  override render() {
+    const currentValue = this.status === 'active' ? this._progress : this.value;
+    const currentStatus = this.getCurrentStatus(currentValue);
+    const helperText = this.getHelperText();
+
+    this._percentage = Math.round((this._progress / this.max) * 100);
+
+    return html`
+      ${this.renderProgressBarLabel(currentStatus)}
+      ${this.renderProgressBar(currentStatus, currentValue)}
+      ${helperText
+        ? html`<div class=${`progress-bar__helper-text ${currentStatus}`}>
+            <h2>${helperText}</h2>
+          </div>`
+        : null}
+    `;
+  }
+
+  private renderProgressBar(
+    currentStatus: ProgressStatus,
+    currentValue: number | null
+  ) {
+    const isIndeterminate = this.status === ProgressStatus.INDETERMINATE;
+    const resolvedValue = [
+      ProgressStatus.SUCCESS,
+      ProgressStatus.ERROR,
+    ].includes(currentStatus as ProgressStatus)
+      ? this.max
+      : currentValue;
+
+    return html`<progress
+      class="${this.getProgressBarClasses(currentStatus)}"
+      max=${this.max}
+      value=${ifDefined(
+        !isIndeterminate && resolvedValue != null ? resolvedValue : undefined
+      )}
+      aria-valuenow=${ifDefined(
+        !isIndeterminate && resolvedValue != null ? resolvedValue : undefined
+      )}
+      aria-valuemin="0"
+      aria-valuemax=${this.max}
+      aria-label=${this.label}
+    ></progress>`;
+  }
+
+  private renderProgressBarLabel(currentStatus: ProgressStatus) {
+    return html`<div class="progress-bar__upper-container">
+      <h2 class="progress-bar__label">
```

Comment:
Should use a `<label>` tag, and associate it with the `<progress>` tag for accessibility.